input/otlp/logs.go

```diff
@@ -109,12 +117,63 @@ func (c *Consumer) convertLogRecord(
 		}
 		event.Span.ID = spanID.HexString()
 	}
-	if attrs := record.Attributes(); attrs.Len() > 0 {
+	attrs := record.Attributes()
+	if isError(record) {
+		event.Processor = model.ErrorProcessor
+		setErrorLabels(attrs, &event)
+		updateErrorType(attrs, &event)
+	} else {
 		setLabels(attrs, &event)
 	}
 	return event
 }
 
+func updateErrorType(m pcommon.Map, event *model.APMEvent) {
+	event.Event.Kind = "event"
+	event.Event.Category = "device"
+	event.Event.Type = "error"
+}
+
+func setErrorLabels(m pcommon.Map, event *model.APMEvent) {
+	var exceptionEscaped bool
+	var exceptionMessage, exceptionStacktrace, exceptionType string
+	var eventName string
+	m.Range(func(k string, v pcommon.Value) bool {
+		switch k {
+		case semconv.AttributeExceptionMessage:
+			exceptionMessage = v.Str()
+		case semconv.AttributeExceptionStacktrace:
+			exceptionStacktrace = v.Str()
+		case semconv.AttributeExceptionType:
+			exceptionType = v.Str()
+		case semconv.AttributeExceptionEscaped:
+			exceptionEscaped = v.Bool()
+		case "event.name":
+			eventName = v.Str()
+		default:
+			setLabel(replaceDots(k), event, ifaceAttributeValue(v))
+		}
+		return true
+	})
+
+	if exceptionMessage != "" || exceptionType != "" {
+		// Per OpenTelemetry semantic conventions:
+		//   `At least one of the following sets of attributes is required:
+		//   - exception.type
+		//   - exception.message`
+		event.Error = convertOpenTelemetryExceptionSpanEvent(
```

Comment:
Should we rename convertOpenTelemetryExceptionSpanEvent (convertOpenTelemetryException?), if it's not specific to span events any more?

Reply:
Perhaps we can get rid of this `setErrorLabels` function, and refactor `convertOpenTelemetryExceptionSpanEvent` so it iterates of a `pcommon.Map` for either logs or span events?

Reply:
Good point! I removed `setErrorLabels` and reused the code in `setLabels`